client/client.go

```diff
@@ -64,3 +74,36 @@ func ValidRedirectURL(rURL *url.URL, redirectURLs []url.URL) (url.URL, error) {
 	}
 	return url.URL{}, ErrorInvalidRedirectURL
 }
+
+func ClientsFromReader(r io.Reader) ([]Client, error) {
+	var c []struct {
+		ID           string   `json:"id"`
+		Secret       []byte   `json:"secret"`
```

Comment:
For --no-db mode, I think this now forces the secrets to be doubly base64 encoded. Once because of the `[]byte` for `encoding/json`, then again for the db code.

* Non base64 encoded secret. Doesn't work on master, still doesn't work on this branch.

```json
[
  {
    "id": "example-app",
    "secret": "example-app-secret",
    "redirectURLs": ["http://127.0.0.1:5555/callback"]
  }
]
```

```
$ ./bin/dex-worker --no-db --clients clients.json
WARN: Running in-process without external database or key rotation
FATAL: Unable to build Server: unable to read clients from file clients.json: illegal base64 data at input byte 7
```

* base64ed secret. Currently works on master, doesn't work on this branch.

```
[
  {
    "id": "example-app",
    "secret": "ZXhhbXBsZS1hcHAtc2VjcmV0",
    "redirectURLs": ["http://127.0.0.1:5555/callback"]
  }
]
```

```
$ ./bin/dex-worker --no-db --clients clients.json
WARN: Running in-process without external database or key rotation
FATAL: Unable to build Server: failed to create client identity repo: illegal base64 data at input byte 16
```

* base64ed version of the base64ed secret. Currently works on master, only thing that works on this branch.

```json
[
  {
    "id": "example-app",
    "secret": "WlhoaGJYQnNaUzFoY0hBdGMyVmpjbVYw",
    "redirectURLs": ["http://127.0.0.1:5555/callback"]
  }
]
```

```
$ ./bin/dex-worker --no-db --clients clients.json
WARN: Running in-process without external database or key rotation
INFO: Loaded IdP connector: id=local type=local
INFO: Loaded IdP connector: id=google type=oidc
INFO: Loaded IdP connector: id=github type=github
INFO: Loaded IdP connector: id=bitbucket type=bitbucket
WARN: bindTemplate not used when searchBeforeAuth specified.
INFO: Loaded IdP connector: id=ldap type=ldap
INFO: Binding to 127.0.0.1:5556...
```